### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by FaridBash**

### CODE REVIEW

File: routes/chatRoutes.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const express=require('express');
+const router=express.Router();
+import {getChatRooms, getChatRoomsByTopic, AddChatRoom } from '../controllers/roomController';
+import {getUsers, getUsersById, setUser} from '../controllers/userController';
+
+router.route('/chatrooms').get(getChatRooms).post(AddChatRoom);
+router.route('/chatrooms/:topic').get(getChatRoomsByTopic);
```
Comment (shellyalmo) [inline routes/chatRoutes.js:7]: @YousefAsilah12 in the DB, is the topic an id or string?  if its mongo id then I think that
@FaridBash should make the route be /:topic_id
for example: "/chatrooms/1234-5332-9865"

**Comment by shellyalmo** [comment]
relates to #3

File: controllers/userController.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+//@route POST /api/user
+//@access Private
+const setUser=( async (req, res)=>{
+    if(!req.body){
+        res.status(400)
+        throw new Error('Please add ID');
+    }
+    const user=await Users.create({
+        name: req.body.name,
+
```
[the remark this answers is not in the document]
Comment (FaridBash) [reply controllers/userController.js:30]: we need yousof to add a property to the model

**Comment by YousefAsilah12** [comment]
done

**Review by shellyalmo — APPROVED** [review]

**Pull request merged**